servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1339,12 +1361,18 @@ public long loadFileInfoFromUfs(TachyonURI path, boolean recursive)
       if (!ufs.exists(ufsPath.getPath())) {
         throw new FileDoesNotExistException(ufsPath.getPath());
       }
-      long ufsBlockSizeByte = ufs.getBlockSizeByte(ufsPath.toString());
-      long fileSizeByte = ufs.getFileSize(ufsPath.toString());
-      // Metadata loaded from UFS has no TTL set.
-      long fileId = create(path, ufsBlockSizeByte, recursive, Constants.NO_TTL);
-      persistFile(fileId, fileSizeByte);
-      return fileId;
+      if (ufs.isFile(ufsPath.getPath())) {
+        long ufsBlockSizeByte = ufs.getBlockSizeByte(ufsPath.toString());
+        long fileSizeByte = ufs.getFileSize(ufsPath.toString());
+        // Metadata loaded from UFS has no TTL set.
+        long fileId = create(path, ufsBlockSizeByte, recursive, Constants.NO_TTL);
+        persistFile(fileId, fileSizeByte);
+        return fileId;
+      } else {
+        InodeTree.CreatePathResult result = mkdir(path, recursive);
+        List<Inode> created = result.getCreated();
+        return created.get(created.size() - 1).getId();
```

Comment:
it seems we have an assumption here the inode corresponding to `path` is the last, which is not really documented anywhere (or I missed it). Is there any way to get the inode for `path` firmly?